uefi-test-runner/src/main.rs

```diff
@@ -58,3 +63,26 @@ fn shutdown(st: &SystemTable) -> ! {
     let rt = st.runtime;
     rt.reset(ResetType::Shutdown, Status::Success, None);
 }
+
+fn qemu_panic_shutdown() -> ! {
```

Comment:
I occasionally like to run these tests on bare metal, so it would be great if there was a way to make this function as a compile-time feature (or more specifically, keep this function, but only write to port `0xF4` when running under QEMU).

Reply:
If you run on the bare metal, you do not have an exit code, so I could use the standard UEFI shutdown method here instead of looping. One more reason for me to pursue my investigation on bundling state with the hook :)

Reply:
You can use the global state we store in the `uefi_services` crate. Considering that UEFI's stdout stops working after exiting boot, I don't see an issue with using standard UEFI exit functions. The user won't see the panic message after `ExitBootServices` anyway. It's the OS's responsibility at that point.

Reply:
That's integrated in the latest version